ragna/deploy/_ui/modal_configuration.py

```diff
@@ -115,12 +134,37 @@ def __init__(self, api_wrapper, **params):
 
         self.got_timezone = False
 
-    def did_click_on_start_chat_button(self, event):
-        if not self.document_uploader.can_proceed_to_upload():
-            self.change_upload_files_label("missing_file")
-        else:
+        self.corpus_or_upload_radiobutton = pn.widgets.RadioButtonGroup.from_param(
+            self.param.corpus_or_upload,
+            button_style="outline",
+            button_type="primary",
+        )
+        self.metadata_filter_rows = None
+
+        self.create_config(components)
+
+    async def did_click_on_start_chat_button(self, event):
+        if self.corpus_or_upload == USE_UPLOAD_LABEL:
+            if not self.document_uploader.can_proceed_to_upload():
+                self.change_upload_files_label("missing_file")
+            else:
+                self.start_chat_button.disabled = True
+                self.document_uploader.perform_upload(
+                    event, self.did_finish_upload, self.corpus_name_input.value
+                )
+
+        elif self.corpus_or_upload == USE_CORPUS_LABEL:
+            if (
```

Comment:
Don't send to API if invalid (and CSS should update to indicate to user that something is wrong).